File: greenbank-spigot/src/main/java/org/flameyosflow/greenbank/commands/noteconomy/GreenbankCommand.java
package org.flameyosflow.greenbank.commands.noteconomy;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.flameyosflow.greenbank.GreenBankMain;
import org.flameyosflow.greenbank.commands.GreenBankCommand;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import static org.flameyosflow.greenbank.utils.MessageUtils.colorMessage;

/**
 * DO NOT MIX THIS UP WITH Green***B***ankCommand
 *
 * @author FlameyosFlow
 */
public class GreenbankCommand extends GreenBankCommand {
    private final GreenBankMain greenBank;

    public GreenbankCommand(GreenBankMain greenBank) {
        this.greenBank = greenBank;
    }

    public void configError(@NotNull CommandSender sender) {
        sender.sendMessage(colorMessage("&cUh oh! something went wrong, please check console!"));
        sender.sendMessage(" ");
        sender.sendMessage(colorMessage("&cMake sure your database configuration is correct!"));
        sender.sendMessage(colorMessage("&cAnd make sure you didn't put any unsupported placeholders in messages.yml's messages!"));
        sender.sendMessage(colorMessage("&4If you think this is not your fault, make sure to report this to the developer."));
    }

    @Override
    public boolean execute(@NotNull CommandSender sender, Command command, String label, String[] args) {
        if (senderDoesNotHavePermission(sender, "greenbank.admin.gb")) {
            sender.sendMessage(colorMessage(NOT_ENOUGH_PERMISSIONS));
            return false;
        }

        if (args.length == 0) {
            sendGreenbankHelpMessage(sender);
        }

        if (args[0].equalsIgnoreCase("reload")) {
            try {
                greenBank.initConfig();
                sender.sendMessage(colorMessage("&aSuccessfully reloaded all GreenBank's configuration files"));
            } catch (IOException error) {
                configError(sender);
            }
        }

        if (args[0].equalsIgnoreCase("version")) {
            sender.sendMessage(colorMessage("&7Greenbank's version is: " + "&c&l" + greenBank.getVersion()));
        }
        return true;
    }
}

File: greenbank-spigot/src/main/java/org/flameyosflow/greenbank/commands/noteconomy/GreenbankCommand.kt
package org.flameyosflow.greenbank.commands.noteconomy

import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.flameyosflow.greenbank.GreenBankMain
import org.flameyosflow.greenbank.commands.GreenBankCommand
import org.flameyosflow.greenbank.utils.MessageUtils
import java.io.IOException

/**
 * DO NOT MIX THIS UP WITH Green***B***ankCommand
 *
 * @author FlameyosFlow
 */
class GreenbankCommand(private val greenBank: GreenBankMain) : GreenBankCommand(greenBank) {
    private fun configError(sender: CommandSender) {
        sender.sendMessage(MessageUtils.colorMessage("&cUh oh! something went wrong, please check console!"))
        sender.sendMessage(" ")
        sender.sendMessage(MessageUtils.colorMessage("&cMake sure your database configuration is correct!"))
        sender.sendMessage(MessageUtils.colorMessage("&cAnd make sure you didn't put any unsupported placeholders in messages.yml's messages!"))
        sender.sendMessage(MessageUtils.colorMessage("&4If you think this is not your fault, make sure to report this to the developer."))
    }

    override fun execute(sender: CommandSender, command: Command?, label: String?, args: Array<String>?): Boolean {
        if (senderDoesNotHavePermission(sender, "greenbank.admin.gb")) {
            sender.sendMessage(MessageUtils.colorMessage(NOT_ENOUGH_PERMISSIONS))
            return false
        }
        when (args!!.size) {
            0 -> sendGreenbankHelpMessage(sender)
            1 -> {
                if (args[0].equals("reload", ignoreCase = true)) {
                    try {
                        greenBank.initConfig()
                        sender.sendMessage(MessageUtils.colorMessage("&aSuccessfully reloaded all GreenBank's configuration files"))
                    } catch (error: IOException) {
                        configError(sender)
                        return false
                    }
                } else if (args[0].equals("version", ignoreCase = true)) {
                    sender.sendMessage(MessageUtils.colorMessage("&7Greenbank's version is: " + "&c&l" + greenBank.version))
                }
            }
        }
        return true
    }
}